plugins/azure/containerapps/containerAppIPRestriction.js

```diff
@@ -0,0 +1,58 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Container Apps IP Restriction Configured',
+    category: 'Container Apps',
+    domain: 'Containers',
+    severity: 'Medium',
+    description: 'Ensures that Container Apps are configured to allow only specific IP addresses.',
+    more_info: 'Azure Container Apps allows you to control inbound traffic by configuring IP ingress restrictions. You can specify rules to either allow traffic only from defined address ranges or deny traffic from specified ranges. Without any IP restriction rules, all inbound traffic is permitted, which may expose your application to security risks.',
+    recommended_action: 'Modify Container Apps and configure IP restriction.',
+    link: 'https://learn.microsoft.com/en-us/azure/container-apps/ip-restrictions',
+    apis: ['containerApps:list'],
+    realtime_triggers: ['microsoftapp:containerapps:write', 'microsoftapp:containerapps:delete'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.containerApps, function(location, rcb) {
+
+            var containerApps = helpers.addSource(cache, source,
+                ['containerApps', 'list', location]);
+
+            if (!containerApps) return rcb();
+
+            if (containerApps.err || !containerApps.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query for Container apps: ' + helpers.addError(containerApps), location);
+                return rcb();
+            }
+
+            if (!containerApps.data.length) {
+                helpers.addResult(results, 0, 'No existing Container apps found', location);
+                return rcb();
+            }
+
+            for (let container of containerApps.data) {
+                if (!container.id) continue;
+                if (container.configuration && container.configuration.ingress && container.configuration.ingress.ipSecurityRestrictions && container.configuration.ingress.ipSecurityRestrictions.length) {
```

Comment:
what happens when third mode is selected. Maybe we should have a setting that checks if an action says allow and ip address is from that list
<img width="522" alt="Screenshot 2024-08-26 at 10 56 03 AM" src="https://github.com/user-attachments/assets/11f9b9b4-8380-4b5b-b051-dcf98e1b11c2">

Reply:
when Allow all traffic is selected ipSecurityRestrictions will be null, and in both cases Allow traffic from IP and deny all other and vice versa ipSecurityRestrictions object will have the list of IP addresses  with status if they are allowed or not .